ceph/ceph_admin/orch.py

```diff
@@ -113,6 +115,48 @@ def check_service(
 
         return False
 
+    def check_service_restart(
+        self,
+        service_name: str,
+        restart_init_time: datetime,
+        timeout: int = 300,
+        interval: int = 5,
+    ) -> bool:
+        """
+        Checks whether the given service was restarted in the previous minute or any other amount of time as specified.
+        Args:
+            service_name (Str): Name of the service
+            restart_init_time (datetime.datetime): Time at which restart was initiated
+            timeout: timeout in seconds
+            interval: interval in seconds
+
+        Returns:
+            True if the service was restarted within the given interval from now, else false
+        """
+        end_time = datetime.now() + timedelta(seconds=timeout)
+
+        while end_time > datetime.now():
```

Comment:
Not sure if we need the `while` statement.

Reply:
Removed while and retries.